### PR TITLE
validate language files against en-US

### PR DESCRIPTION
**Pull request by kevinkreiser**
this works really nice in that it tells both what and where the problem is. i tested missing phrases, missing replacements, missing tags, wrong names of phrases (wrong keys), wrong items under the instruction, and if the posix locale is supported. only thing i dont like is that its a unit test. id rather it be run in make and fail compilation instead but this is probably too strict anyway.

oh also it found a bug in german language file

@dgearhart please have a look

### CODE REVIEW

File: locales/de-DE.json [hunk 1]
```diff
@@ -302,7 +302,7 @@
         "0": "Geradeaus über die Rampe fahren.",
         "1": "Auf <BRANCH_SIGN> geradeaus über die Rampe fahren.",
         "2": "Geradeaus über die Rampe fahren Richtung <TOWARD_SIGN>.",
-        "4": "Geradeaus über die Rampe <NAME_SIGN> fahren."
+        "3": "Geradeaus über die Rampe <NAME_SIGN> fahren."
```
Comment (kevinkreiser) [inline locales/de-DE.json]: test found this bug

File: src/odin/util.cc [hunk 2]
```diff
@@ -183,5 +181,9 @@ const locales_singleton_t& get_locales() {
   return locales;
 }
 
+const std::unordered_map<std::string, std::string>& get_locales_json() {
```
Comment (kevinkreiser) [inline src/odin/util.cc]: didnt like having to add this but was the best option for testability

File: src/odin/util.cc [hunk 3]
```diff
@@ -138,8 +138,6 @@ std::string get_localized_time(const std::string& date_time, const std::string&
       output_facet = new boost::posix_time::time_facet("%l:%M %p");
       ss.imbue(std::locale(std::locale::classic(), output_facet));
       ss << pt;
-      std::cout << ss.str() << std::endl;
```
Comment (kevinkreiser) [inline src/odin/util.cc]: @gknisely you probably meant to remove this at some point but forgot?

Comment (gknisely) [inline src/odin/util.cc]: yes thank you

**Comment by gknisely** [comment]
:ship:

**Comment by dgearhart** [comment]
:shipit:

**Pull request merged**